model/adjuster/bad_span_references.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package adjuster
+
+import (
+	"fmt"
+
+	"github.com/jaegertracing/jaeger/model"
+)
+
+// SpanReferences creates am adjuster that removes invalid span references, e.g. with traceID==0
+func SpanReferences() Adjuster {
+	return Func(func(trace *model.Trace) (*model.Trace, error) {
+		adjuster := &spanReferenceAdjuster{}
+		for _, span := range trace.Spans {
+			adjuster.adjust(span)
+		}
+		return trace, nil
+	})
+}
+
+type spanReferenceAdjuster struct{}
+
+func (s *spanReferenceAdjuster) adjust(span *model.Span) *model.Span {
```

Comment:
nit: I don't actually see any reason for these functions to be defined as pointer methods